packages/react-native-client/src/hooks/useFishjamEvent.ts

```diff
@@ -16,6 +16,20 @@ export const ReceivableEvents = {
   Reconnected: 'Reconnected',
 } as const;
 
-export const eventEmitter = new EventEmitter(
+const eventEmitter = new EventEmitter(
   RNFishjamClientModule ?? NativeModulesProxy.RNFishjamClient,
 );
+
+export function useFishjamEvent<T>(
+  eventName: keyof typeof ReceivableEvents,
+  callback: (event: T) => void,
+) {
+  useEffect(() => {
+    const eventListener = eventEmitter.addListener<
+      Record<keyof typeof ReceivableEvents, T>
+    >(eventName, (event) => {
+      callback(event[eventName]);
```

Comment:
**issue:** If it's a hook, why use a callback, instead of `useState`? It would be a more React way 🤔 
And from the way I see it's used like that anyway.

Reply:
Nevermind. I misunderstood, how it works....

Reply:
There various ways on how these hooks were used. It would be possible to refactor code to keep state inside it - but that might be next step in refactor at some point.